src/map/battle.c

```diff
@@ -3231,11 +3231,10 @@ static int64 battle_calc_damage(struct block_list *src, struct block_list *bl, s
 		if (!damage) return 0;
 
 		if( (sce = sc->data[SC_LIGHTNINGWALK]) && flag&BF_LONG && rnd()%100 < sce->val1 ) {
-			int dx[8]={0,-1,-1,-1,0,1,1,1};
-			int dy[8]={1,1,0,-1,-1,-1,0,1};
-			uint8 dir = map->calc_dir(bl, src->x, src->y);
-			if( unit->movepos(bl, src->x-dx[dir], src->y-dy[dir], 1, 1) ) {
-				clif->slide(bl,src->x-dx[dir],src->y-dy[dir]);
+			enum unit_dir dir = map->calc_dir(bl, src->x, src->y);
+			Assert_ret(dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX);
```

Comment:
I'd prefer to go the safe way and check this even when return values of map->calc_dir are used for array-indeces, in case someone will ever fiddle with map_calc_dir

Reply:
@hemagx resolve conservation if you accept this answer.